### DISCUSSION TITLE
开源教程WTF Solidity

### DISCUSSION THREAD

**Discussion by AmazingAng** [AmazingAng/WTF-Solidity#6] [Announcements]
大家好，
这两天我整理了下WTF Solidity极简教程的github repo，把所有的文字教程也加进来了。
之后对教程有问题的，可以直接提issue，或者改正后提PR（提1次PR可以进核心群）。
让我们一起建立Web3最好的Solidity教程和技术学习社区！LFG

by 0xAA

**Comment by kan180** [comment]
大佬，学习了你写的solidity教程，里面有一个问题，也是我自己一直没理解的地方，请大佬赐教。BAYC的主合约里 provenance这个值要怎么用，是不是在合约部署后需要SET一次BAYC_PROVENANCE值。怎么拼这个串呢？

**Comment by AmazingAng** [comment]
provenance是所有图片的hash拼到一起再hash一下，合约部署后/开盲盒前set

**Reply by kan180** [reply]
是为了做一致性校验吗？这个hash算法可以自己随便定义吧？

**Reply by AmazingAng** [reply]
对。原意是在开图之前证明项目方没有换图（做恶），但其实没什么卵用。

**Reply by kan180** [reply]
明白了，确实没啥大用，项目方可以反复set，多谢指教了。
